docs/src/app/(public)/(content)/react/handbook/composition/page.mdx

```diff
@@ -0,0 +1,79 @@
+# Composition
+
+<Subtitle>A guide to composing Base UI components with your own React components.</Subtitle>
+<Meta
+  name="description"
+  content="A guide to composing Base UI components with your own React components."
+/>
+
+### Composing custom React components
+
+The `render` prop lets you override a component's rendered element with your own React component. For example, `Menu.Trigger` renders a `<button>` by default. The code snippet below shows how to use a custom component instead.
+
+```tsx title="index.tsx"
+<Menu.Trigger render={<MyFancyButton size="md" />} />
+```
+
+The custom component must forward a `ref`, and spread all the received props, to its underlying DOM node.
+
+### Composing multiple components
+
+In situations where you need to compose multiple Base UI components with custom React components, `render` props can be nested as deeply as necessary. Working with Tooltip is a common example.
+
+```tsx title="index.tsx"
+<Dialog.Root>
+  <Tooltip.Root>
+    <Tooltip.Trigger
+    {/* prettier-ignore */}
+      render={
+        <Dialog.Trigger
+          render={
+            <MyFancyButton size="md" />
+          }
+        />
+      }
+    />
+    <Tooltip.Portal>...</Tooltip.Portal>
+  </Tooltip.Root>
+  <Dialog.Portal>...</Dialog.Portal>
+</Dialog.Root>
+```
+
+### Changing the default rendered element
+
+You can also use the `render` prop to override the rendered element of the component. For example, `Menu.Item` renders a `<div>` by default.
+The code snippet below shows how to render an `<a>` instead.
```

Comment:
```suggestion

 ### Changing the default rendered element

 You can also use the `render` prop to override the rendered element of the component.
 
 For example, you may render a menu item as an `<a>` element so that it works like a link.
```